src/frontend/screens/LeaveProject/LeaveProject.tsx

```diff
@@ -0,0 +1,200 @@
+// @ts-check
+import React, { useContext, useState } from "react";
+import { defineMessage, FormattedMessage } from "react-intl";
+import { View, Text, StyleSheet, Image } from "react-native";
+import CheckBox from "@react-native-community/checkbox";
+import ConfigContext from "../../context/ConfigContext";
+import { useMemo } from "react";
+import Button from "../../sharedComponents/Button";
+import { useNavigation } from "react-navigation-hooks";
+import {
+  NavigationStackOptions,
+  NavigationStackScreenComponent,
+} from "react-navigation-stack";
+import HeaderTitle from "../../sharedComponents/HeaderTitle";
+import ObservationHeaderRight from "../Observation/ObservationHeaderRight";
+import AppStack from "../../AppStack";
+
+const m = defineMessage({
+  leaveProjectTitle: {
+    id: "screens.LeaveProject.LeaveProject.leaveProjectTitle",
+    defaultMessage: "Leave Project{projectName}?",
+  },
+  headerTitle: {
+    id: "screens.LeaveProject.LeaveProject.headerTitle",
+    defaultMessage: "Leave Project",
+  },
+  willDelete: {
+    id: "screens.LeaveProject.LeaveProject.willDelete",
+    defaultMessage:
+      "This will delete {numOfObservations} observations and {numOfPics} Pictures",
+  },
+  agreeToDelete: {
+    id: "screens.LeaveProject.LeaveProject.agreeToDelete",
+    defaultMessage: "I understand I will be deleting all data from my device.",
+  },
+  leaveButton: {
+    id: "screens.LeaveProject.LeaveProject.leaveButton",
+    defaultMessage: "Leave Project",
+  },
+  cancel: {
+    id: "screens.LeaveProject.LeaveProject.cancel",
+    defaultMessage: "Cancel",
+  },
+  confirmDelete: {
+    id: "screens.LeaveProject.LeaveProject.confirmDelete",
+    defaultMessage: "Please check the box to confirm",
+  },
+  syncWarning: {
+    id: "screens.LeaveProject.LeaveProject.syncWarning",
+    defaultMessage:
+      "Sync with a team member so you don't loose all your observations.",
+  },
+});
+
+const navOptions: NavigationStackOptions = {
+  headerTitle: () => (
+    <HeaderTitle>
+      <FormattedMessage {...m.headerTitle} />
+    </HeaderTitle>
+  ),
+};
+
+export const LeaveProjectScreen: NavigationStackScreenComponent = () => {
+  const [confirmDelete, setConfirmDelete] = React.useState(false);
+  const [pristine, setPristine] = useState(true);
```

Comment:
this feels like an odd variable name. not really sure what it means in this context

Reply:
Pristine vs dirty. So a pristine checkbox means it hasnt been touched, therefore it shouldnt show an error. But once it has been touched, it has been made dirty, so therefore show an error. It avoids having an error right away when the user hasnt done anything yet

Reply:
ah ok. is there a different name we could use then? maybe `checkboxTouched` or `touched`? `pristine` is not super intuitive for me personally

Reply:
It would change the logic if I used touched. I can use "untouched"?

Reply:
sure 👍